src/tool/subcommands/api_cmd.rs

```diff
@@ -624,9 +627,23 @@ async fn compare_apis(
 
     tests.sort_by_key(|test| test.request.method_name);
 
-    let mut results = HashMap::default();
+    run_tests(tests, &forest, &lotus, filter, fail_fast, run_ignored).await
+}
 
+async fn run_tests(
+    tests: Vec<RpcTest>,
+    forest: &ApiInfo,
+    lotus: &ApiInfo,
+    filter: String,
+    fail_fast: bool,
+    run_ignored: RunIgnored,
+) -> anyhow::Result<()> {
+    let (tx, mut rx) = mpsc::unbounded_channel();
```

Comment:
Did you consider using `FuturesUnordered` instead? I'd expect the code with it to be a bit more straightforward than with channels.

Reply:
right, updated the code to use `FuturesUnordered`